src/main/java/bio/terra/common/exception/InconsistentFieldsException.java

```diff
@@ -0,0 +1,11 @@
+package bio.terra.common.exception;
+
+public class InconsistentFieldsException extends BadRequestException {
```

Comment:
I wouldn't have thought this one would be common, but OK.
We probably need better documentation for what these are supposed to mean.
No necessarily as part of this PR.

Reply:
Updated comment and leaving here, per discussion.

Reply:
My original intent for this exception was to mean we got an API request that is internally inconsistent. The one case is where the caller tries to assign a user to a private resource.

I see I used it a few other places that should probably be logic exceptions - 500s - not InconsistentFieldsExceptions.